### PR TITLE
add flux-sched container tutorial

### PR DESCRIPTION
**Pull request by vsoch**
Problem: the easiest way to get started with flux is via the container, but we do not have an obvious tutorial. It is hidden (and short) inside of the quickstart pages.
Solution: create a set of container tutorials, to which we can also add using Flux with docker-compose.

### CODE REVIEW

**Comment by vsoch** [comment]
Preview: https://flux-framework--217.org.readthedocs.build/en/217/tutorials/containers/flux-sched-container.html

File: tutorials/containers/flux-sched-container.rst [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+----------------------
+
+Run the container as follows:
+
+.. code-block:: bash
+
+  # This says "run an interactive terminal"
+  $ docker run -it fluxrm/flux-sched:latest
+
+At this point, even before starting an instance, we can inspect what is going on with Flux.
```
Comment (grondo) [inline tutorials/containers/flux-sched-container.rst:31]: The entrypoint for all the docker images is `/usr/bin/flux start /bin/bash` so a test instance is already started.

Maybe reword to say something like "By default, the docker container starts a test Flux instance of size 1"

File: tutorials/containers/flux-sched-container.rst [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+----------------------
+Starting the Container
+----------------------
+
+Run the container as follows:
+
+.. code-block:: bash
+
+  # This says "run an interactive terminal"
```
Comment (grondo) [inline tutorials/containers/flux-sched-container.rst:28]: in reference to the other comment
```suggestion
  # This says "start an interactive shell under a test Flux instance of size=1"
```

File: tutorials/containers/flux-sched-container.rst [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+  $ docker run -it fluxrm/flux-sched:latest
+
+At this point, even before starting an instance, we can inspect what is going on with Flux.
+
+
+.. code-block:: shell
+
+  # What is the size of resources we have?
```
Comment (grondo) [inline tutorials/containers/flux-sched-container.rst:36]:
```suggestion
  # What is the size of the current Flux instance (i.e. number of flux-broker processes)
```

File: tutorials/containers/flux-sched-container.rst [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+  $ flux getattr size
+  1
+
+  # Inspect networking
+  $ flux overlay status
+
+  # What resources do we have, and what are their statuses?
```
Comment (grondo) [inline tutorials/containers/flux-sched-container.rst:43]:
```suggestion
  # What resources do we have, and what are their states?
```

File: tutorials/containers/flux-sched-container.rst [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+
+
+.. code-block:: shell
+
+  # What is the size of resources we have?
+  $ flux getattr size
+  1
+
+  # Inspect networking
+  $ flux overlay status
```
Comment (grondo) [inline tutorials/containers/flux-sched-container.rst:41]: This isn't too interesting for an instance of size=1, so it could be dropped.

File: tutorials/containers/flux-sched-container.rst [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+.. code-block:: shell
+
+  # Where are different Flux things located (the environment)
+  $ flux env
+
+------------
+Running Jobs
+------------
+
+And submit a job to get back an id. The "submit" command is headless, and "run" is interactive:
```
Comment (grondo) [inline tutorials/containers/flux-sched-container.rst:65]:
```suggestion
``flux submit`` submits a job which will be scheduled and run in the background. It prints the jobid to standard output upon successful submission. To run a job interactively, use ``flux run`` which submits and then attaches to the job, displays job output in real time, and does not exit until the job finishes.
```

File: tutorials/containers/flux-sched-container.rst [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+
+    $ flux submit hostname
+    ƒM5k8m7m
+    $ flux run hostname
+    f9004106893d
+
+You can inspect ``--help`` for each of the commands above to see it's possible
+to customize the number of nodes, tasks, cores, and other variables.
+There are many different options to customize your job submission. For further
+details, please see :core:man1:`flux-submit`.
```
Comment (grondo) [inline tutorials/containers/flux-sched-container.rst:77]:
```suggestion
details, please see :core:man1:`flux-submit` or run ``flux help submit``.
```

File: tutorials/containers/flux-sched-container.rst [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+
+The identifier shown above is a :ref:`jobid<fluid>` (e.g., ``f9004106893d``). This kind of identifer
```
Comment (grondo) [inline tutorials/containers/flux-sched-container.rst:79]: ``f9004106893d`` appears to be the hostname. ``ƒM5k8m7m`` is the jobid.

File: tutorials/containers/flux-sched-container.rst [hunk 9]
```diff
@@ -0,0 +1,232 @@
...
+
+
+.. code-block:: shell
+
+    $ flux jobs
+           JOBID USER     NAME       ST NTASKS NNODES     TIME INFO
+        ƒq3d755R fluxuser sleep       R      1      1   2.811s 848fc387afd7
+        ƒpcByPgK fluxuser sleep       R      1      1   3.805s 848fc387afd7
+
+But wait, what happened to our first two jobs? Add an ``-a`` for "all" to see all the jobs.
```
Comment (grondo) [inline tutorials/containers/flux-sched-container.rst:96]:
```suggestion
But wait, what happened to our first two jobs? ``flux jobs`` only shows "active" jobs by default, add an ``-a`` for "all" to see all the jobs.
```

File: tutorials/containers/flux-sched-container.rst [hunk 10]
```diff
@@ -0,0 +1,232 @@
...
+           JOBID USER     NAME       ST NTASKS NNODES     TIME INFO
+        ƒq3d755R fluxuser sleep       R      1      1   39.98s 848fc387afd7
+        ƒpcByPgK fluxuser sleep       R      1      1   40.97s 848fc387afd7
+        ƒDi2hxdm fluxuser hostname   CD      1      1   0.019s 848fc387afd7
+        ƒBmTaVZ9 fluxuser hostname   CD      1      1   0.025s 848fc387afd7
+
+.. note::
+
+   See if you can figure out how to list jobs by a particular status, e.g., ``R`` in the output 
+   above means "running." Try ``flux jobs --help``.
```
Comment (grondo) [inline tutorials/containers/flux-sched-container.rst:111]:
```suggestion
   above means "running." Try ``flux jobs --help`` or ``flux help jobs``.
```

File: tutorials/containers/flux-sched-container.rst [hunk 11]
```diff
@@ -0,0 +1,232 @@
...
+
+    $ flux job attach ƒ4evXWb9Z
+    Hello I am loop iteration 0.
+    Hello I am loop iteration 1.
+    Hello I am loop iteration 2.
+    Hello I am loop iteration 3.
+    Hello I am loop iteration 4.
+    Hello I am loop iteration 5.
+
+See :core:man1:`flux-job` for more information on ``flux job attach``.
```
Comment (grondo) [inline tutorials/containers/flux-sched-container.rst:147]:
```suggestion
See ``flux help job`` or :core:man1:`flux-job` for more information on ``flux job attach``.
```

File: tutorials/containers/flux-sched-container.rst [hunk 12]
```diff
@@ -0,0 +1,232 @@
...
+
+
+------------
+Viewing Jobs
+------------
+
+Aside from listing jobs with ``flux jobs`` there are other ways to get metadata about jobs.
+For your running jobs, you can use ``flux pstree`` to see exactly that - a tree of processes.
```
Comment (grondo) [inline tutorials/containers/flux-sched-container.rst:155]:
```suggestion
For your running jobs, you can use ``flux pstree`` to see exactly that - a tree of jobs.
```

File: tutorials/containers/flux-sched-container.rst [hunk 13]
```diff
@@ -0,0 +1,232 @@
...
+        "inactive": 0,
+        "total": 3
+      },
+      "successful": 0,
+      "failed": 0,
+      "canceled": 0,
+      "timeout": 0,
+      "inactive_purged": 0,
+      "queues": []
+    }
```
Comment (grondo) [inline tutorials/containers/flux-sched-container.rst:208]: `flux job stats` doesn't display information about an individual job, it displays job statistics for the instance as a whole. This is the plumbing equivalent of `flux jobs --stats-only`, so maybe we should drop this section?

File: tutorials/containers/flux-sched-container.rst (hunk 5 above)
Comment (vsoch) [reply tutorials/containers/flux-sched-container.rst:41]: okay will drop - I'll add this to the docker-compose example since that might be more interesting.

File: tutorials/containers/flux-sched-container.rst (hunk 13 above)
Comment (vsoch) [reply tutorials/containers/flux-sched-container.rst:208]: okay I can drop! High level dinosaur note  here - as a user I'm always looking for a `flux job <id>` (something) that gives me a json dump of everything I know about a job. I know it's possible with the Python API but I haven't found a good equivalent here. Maybe just `flux job <jobid> --json` or something could do it?

**Comment by vsoch** [comment]
Okay I think I got all of them - new preview: https://flux-framework--217.org.readthedocs.build/en/217/tutorials/containers/flux-sched-container.html

File: tutorials/containers/flux-sched-container.rst (hunk 13 above)
Comment (grondo) [reply tutorials/containers/flux-sched-container.rst]: Good idea! I had thought we'd had an issue open on this already, but I cannot find it! The obvious choice might be `flux job info --all JOBID`, which would dump all known info "keys" in a single JSON object.

**Review by grondo — APPROVED** [review]
LGTM! Thanks.
